pkg/trtl/honu.go

```diff
@@ -108,8 +112,179 @@ func (h *HonuService) Delete(ctx context.Context, in *pb.DeleteRequest) (out *pb
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
+// Iter is a unary request to fetch a materialized collection of key/value pairs based
+// on a shared prefix. If no prefix is specified an entire namespace may be returned.
+// This RPC supports pagination to ensure that replies do not get to large. The default
+// page size is 100 items, though this can be modified in the options. The next page
+// token in the result will contain the next page to request, or will be empty if there
+// are no more results to be supplied.
+//
+// Note that there are no snapshot guarantees during iteration, meaning that if the
+// underlying database changes between requests, these changes could be reflected during
+// iteration. For snapshot isolation in iteration, use the Cursor RPC.
+//
+// There are several options that modulate the Iter response:
+//   - return_meta: each key/value pair will contain the object metadata
+//   - iter_no_keys: each key/value pair will not have a key associated with it
+//   - iter_no_values: each key/value pair will not have a value associaed with it
+//   - page_token: the page of results that the user wishes to fetch
+//   - page_size: the number of results to be returned in the request
 func (h *HonuService) Iter(ctx context.Context, in *pb.IterRequest) (out *pb.IterReply, err error) {
-	return nil, status.Error(codes.Unimplemented, "not implemented")
+	// Ensure the namespace is not reserved
+	if _, found := reservedNamespaces[in.Namespace]; found {
+		log.Warn().Str("namespace", in.Namespace).Msg("cannot use reserved namespace")
+		return nil, status.Error(codes.PermissionDenied, "cannot used reserved namespace")
+	}
+
+	// Load the options from the request
+	var opts *pb.Options
+	if in.Options != nil {
+		opts = in.Options
+	} else {
+		// Create empty options
+		opts = &pb.Options{}
+	}
+
+	// Load defaults into options
+	if opts.PageSize == 0 {
+		opts.PageSize = 100
+	}
```

Comment:
We should probably create a helper function that instantiates default options then updates the user options, I think this would be useful in all of our RPC functions. Let me know what you guys think!